source/common/stats/stats_impl.cc

```diff
@@ -290,6 +297,9 @@ void HeapRawStatDataAllocator::free(RawStatData& data) {
 
 void RawStatData::initialize(absl::string_view key) {
   ASSERT(!initialized());
+  if (key.find('\0') != std::string::npos) {
+    throw EnvoyException(fmt::format("Stat names may not contain NULL ({})", key));
```

Comment:
There are certain cases that this won't cover. For example the "x-envoy-upstream-alt-stat-name" header. I'm mainly wondering if we should be sanitizing at a higher level? I understand why you want to do it here, but it still feels potentially like the wrong place given that we can for sure wind up in this situation at runtime in at least that case and possible others?

Would it be better to sanitize embedded nulls and replace them with underscore or something? Just throwing it out there.

Reply:
Are you saying `x-envoy-upstream-alt-stat-name` will cause an issue because we throw an exception mid-request? I think we still go through this code path in that scenario, but maybe I've missed something.

If the former, sure, we can do a replace operation like we do for `:`.

Reply:
We will throw an exception in the request path, which we definitely do not handle correctly, and will end up crashing. Currently that is only possible for internal requests since external requests the header is sanitized, but still seems pretty bad. I'm mostly just thinking of ways to sanitize and I wonder if we just do a replace to "fix" the issue and call it good?